src/session_impl.cpp

```diff
@@ -3732,11 +3732,56 @@ namespace aux {
 	}
 
 	namespace {
-		bool last_optimistic_unchoke_cmp(torrent_peer const* const l
-			, torrent_peer const* const r)
+		struct opt_unchoke_candidate
 		{
-			return l->last_optimistically_unchoked
-				< r->last_optimistically_unchoked;
+			opt_unchoke_candidate(boost::shared_ptr<peer_connection> const* tp
+				, std::vector<boost::shared_ptr<plugin>>* ps)
+				: peer(tp)
+				, plugins(ps)
+				, ext_priority(std::numeric_limits<uint64_t>::max())
```

Comment:
in a C++11 world, I think simple initialization of members should be done at member declarations (it's just more robust in general)